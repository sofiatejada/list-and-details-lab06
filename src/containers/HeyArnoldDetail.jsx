import React, { useState, useEffect } from 'react';
import { fetchOneCharacter } from '../services/heyArnoldApi';
import spinner from './heyarnoldspinner.gif';
import { Link, useParams } from 'react-router-dom';
import Character from '../components/characters/Character';

const HeyArnoldDetail = () => {

  const { id } = useParams();
  const [character, setCharacter] = useState({});
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    fetchOneCharacter(id)
      .then((character) => setCharacter(character))
      .finally(() => setLoading(false));
  }, [id]);

  return (
    <main>
      <header>
        <Link to="/">
          Homepage
        </Link>
      </header>
      <content>
        {loading ? 
          <img src={spinner} alt="loading gif" /> : 
          <Character {...character} />}
      </content>
    </main>
  );
};

export default HeyArnoldDetail;
