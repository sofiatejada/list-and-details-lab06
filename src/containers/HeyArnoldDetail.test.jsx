import React from 'react';
import { render, screen } from '@testing-library/react';
import HeyArnoldDetail from './HeyArnoldDetail';
import { MemoryRouter } from 'react-router-dom';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import detail from './detail.json';

const server = setupServer(
  // eslint-disable-next-line max-len
  rest.get('https://hey-arnold-api.herokuapp.com/api/v1/characters/:id', 
    (req, res, ctx) => {
      return res(
        ctx.json(detail)
      );
    })
);

describe('list', () => {

  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('displays one hey arnold characters', async () => {
    render(
      <MemoryRouter initialEntries={['/5da237699734fdcb7bef8f5c']}>
        <HeyArnoldDetail />
      </MemoryRouter>
    );

    screen.getAllByAltText('loading gif');
    const item = await screen.findByLabelText('figure', {}, { timeout: 5000 });
    screen.debug();
    expect(item).toMatchSnapshot();
  });
});
