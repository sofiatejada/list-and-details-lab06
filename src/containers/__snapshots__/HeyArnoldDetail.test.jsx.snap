// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list displays one hey arnold characters 1`] = `
<figure
  aria-label="figure"
>
  <img
    alt="image of Nancy"
    src="https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
  />
  <figcaption>
    Nancy
  </figcaption>
</figure>
`;
